libcasr/src/sarif.rs

```diff
@@ -0,0 +1,404 @@
+        map.insert("runs".to_string(), Value::Array(runs));
+        Self {
+            json: Value::Object(map),
+            id: 0,
+        }
+    }
+
+    /// Set name for SARIF tool:driver.
+    /// NOTE: before use this method,
+    /// use SarifReprot::new() to get report.
+    ///
+    ///  # Arguments
+    ///
+    ///  * 'name' - tool:driver name (default is CASR)
+    pub fn set_name(&mut self, name: &str) {
+        let driver = self.json.as_object_mut().unwrap()["runs"]
+            .as_array_mut()
+            .unwrap()[0]
+            .as_object_mut()
+            .unwrap()["tool"]
+            .as_object_mut()
+            .unwrap()["driver"]
+            .as_object_mut()
+            .unwrap();
+        driver.insert("name".to_string(), Value::String(name.to_string()));
+    }
+
+    /// SARIF rule from ExecutionClass.
+    /// NOTE: before using this method,
+    /// use SarifReprot::new() to get report.
+    ///
+    ///  # Arguments
+    ///
+    ///  * 'class' - ExecutionClass from CrashReport
+    ///
+    ///  # Return
+    ///
+    ///  If Rule is new, than it's json object is returned and ruleId
+    ///  else ruleId only is returned.
+    fn rule(&self, class: &ExecutionClass) -> (Option<Value>, String) {
+        let rule_id = if let Some(pos) = CLASSES
+            .iter()
+            .position(|item| item.1 == class.short_description)
+        {
+            format!("F{pos}")
+        } else {
+            let s = format!("G{}", self.id);
+            s
+        };
+
+        let rules = self.json.as_object().unwrap()["runs"].as_array().unwrap()[0]
+            .as_object()
+            .unwrap()["tool"]
+            .as_object()
+            .unwrap()["driver"]
+            .as_object()
+            .unwrap()["rules"]
+            .as_array()
+            .unwrap();
+        if rules
+            .iter()
+            .any(|r| r.as_object().unwrap()["name"].as_str().unwrap() == class.short_description)
+        {
+            // ExecutionClass is already added.
+            return (None, rule_id);
+        }
+        let mut rule = Map::new();
+        rule.insert("id".to_string(), Value::String(rule_id.clone()));
+        rule.insert(
+            "name".to_string(),
+            Value::String(class.short_description.clone()),
+        );
+        let mut short_desc = Map::new();
+        short_desc.insert("text".to_string(), Value::String(class.description.clone()));
+        rule.insert(
+            "shortDescription".to_string(),
+            Value::Object(short_desc.clone()),
+        );
+        let mut full_desc = Map::new();
+        full_desc.insert("text".to_string(), Value::String(class.explanation.clone()));
+        rule.insert(
+            "fullDescription".to_string(),
+            Value::Object(full_desc.clone()),
+        );
+        let mut properties = Map::new();
+        properties.insert(
+            "problem.severity".to_string(),
+            Value::String(class.severity.clone()),
```

Comment:
Can be one of: error, warning, or recommendation.

Reply:
Use properties.security-severity:

```
(Recommended) A string representing a score that indicates the level of severity, between 0.0 and 10.0, for
security queries (@tags includes security). This, with the properties.precision property, determines whether
the results are displayed by default on GitHub so that the results with the highest security-severity, and
highest precision are shown first. Code scanning translates numerical scores as follows: over 9.0 is critical,
7.0 to 8.9 is high, 4.0 to 6.9 is medium and 3.9 or less is low.
```